addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -311,3 +315,58 @@ def _get_iap_search_term(self, email):
         """
         domain = tools.email_domain_extract(email)
         return ("@" + domain) if domain not in iap_tools._MAIL_DOMAIN_BLACKLIST else email
+
+    def _prepare_translations(self, plugin):
+        """
+        Returns translations for terms used in the mail addons, this method can be overridden by other modules to return
+        their related translations.
+
+        for params inside the translated string, only named string params e.g: %(param)s are supported by the plugins,
+        unnamed params like %s are not supported.
+
+        each entry must have the exact same values for it's key and it's translation value in the returned dictionary.
+        e.g: "Company created": _("Company created") is correct and "company_created": ("Company created") is incorrect,
+        this allows for easier and more intuitive manipulation of translated strings from the plugins side.
+
+        :param plugin: name of the plugin asking for translations
+        """
+        translations = {}
+
+        if plugin == "outlook":
+            translations = {
+                "Logout": _("Logout"),
+                "Company created": _("Company created"),
```

Comment:
Isn't it possible to use the source callable (did not use it since a while, maybe check with @mart-e that is translation master), and do something like
```
return dict(
    ((raw_string), translate(raw_string))
    for raw_string in ['str1', 'str2']
)
```

It allow to avoid somehow defining twice the same sentence (hence limiting typo risk).

Reply:
But if we put the translation in the XML, how can we get it in the controller ?

Reply:
And the route `/web/webclient/translations/` will return all translation of the module, even those we do not need ?

Reply:
I don't think that putting translations in an xml file would solve the issue, as we will also have to write strings manually like we are doing here.

Reply:
@nounoubensebia you indeed still need to write the strings at least once (we can't magically guess it), this discussion was how to avoid writing it twice :) @std-odoo getting a bit more web translations is not the end of the word if it's compensated with easier code to maintain or cache system (as the existing route does with the translation hash, but I guess outlook doesn't have a cache system). Anyway, it was a random idea, it always depends on how frequently this is going to be called.

Reply:
@mart-e Ah okay I see, but I don't quite understand how we can avoid having to write this twice if we use an XML file? and also will strings get automatically exported for translations like when we use python code?